odpf/siren/siren.proto

```diff
@@ -0,0 +1,333 @@
+}
+
+message ListAlertHistoryResponse {
+    repeated AlertHistory alerts = 1;
+}
+
+message AlertHistory {
+    uint64 id = 1;
+    string name = 2;
+    string template_id = 3;
+    string metric_name = 4;
+    string metric_value = 5;
+    string level = 6;
+    string created_at = 7;
+    string updated_at = 8;
+}
+
+message Annotations {
+    string metric_name = 1;
+    string metric_value = 2;
+    string resource = 3;
+    string template = 4;
+}
+
+message Labels {
+    string severity = 1;
+}
+
+message Alerts {
+    Annotations annotations = 1;
+    Labels labels = 2;
+    string status = 3;
+}
+
+message CreateAlertHistoryRequest {
+    repeated Alerts alerts = 1;
+}
+
+message CreateAlertHistoryResponse {
+    repeated AlertHistory alerts = 1;
+}
+
+message Workspace {
+    string id = 1;
+    string name = 2;
+}
+
+message ListWorkspaceChannelsRequest {
+    string workspace_name = 1;
+}
+
+message ListWorkspaceChannelsResponse {
+    repeated Workspace data = 1;
+}
+
+message ExchangeCodeRequest {
+    string code = 1;
+    string workspace = 2;
+}
+
+message ExchangeCodeResponse {
+    bool ok = 1;
+}
+
+message GetAlertCredentialsRequest{
+    string team_name = 1;
+}
+
+message Critical {
+    string channel = 1;
+}
+
+message Warning {
+    string channel = 1;
+}
+
+message SlackConfig {
+    Critical critical = 1;
+    Warning warning = 2;
+}
+
+message GetAlertCredentialsResponse {
+    string entity = 1;
+    string team_name = 2;
+    string pagerduty_credentials = 3;
+    SlackConfig slack_config = 4;
+}
+
+message UpdateAlertCredentialsRequest {
+    string entity = 1;
+    string team_name = 2;
+    string pagerduty_credentials = 3;
+    SlackConfig slack_config = 4;
+}
+message UpdateAlertCredentialsResponse{}
+
+
+message SendSlackNotificationRequest{
+    string provider = 1;
+    string entity = 2;
+    string message = 3;
+    string receiver_name = 4;
+    string receiver_type = 5;
+    repeated google.protobuf.Struct blocks = 6;
+}
+
+message SendSlackNotificationResponse {
+    bool ok = 1;
+}
+
+message ListRulesRequest {
+    string namespace = 1;
+    string entity= 2;
+    string group_name = 3;
+    string status = 4;
+    string template = 5;
+}
+
+message Variables {
+    string name = 1;
+    string type = 2;
+    string value = 3;
+    string description = 4;
+}
+
+message Rule {
+    uint64 id = 1;
+    string name = 2;
+    string entity = 3;
+    string namespace = 4;
+    string group_name = 5;
+    string template = 6;
+    string status = 7;
+    google.protobuf.Timestamp created_at = 8;
+    google.protobuf.Timestamp updated_at = 9;
+    repeated Variables variables = 10;
+}
+
+message ListRulesResponse {
+    repeated Rule rules = 1;
+}
+
+message UpdateRuleRequest {
+    uint64 id = 1;
+    string name = 2;
+    string entity = 3;
+    string namespace = 4;
+    string group_name = 5;
+    string template = 6;
+    string status = 7;
+    repeated Variables variables = 10;
```

Comment:
We can put a field with `Rule` type to avoid repeating Rule's fields
```suggestion
    Rule rule = 1;
```

And do this in the service method to keep the request body:
```
    rpc UpdateRule(UpdateRuleRequest) returns (Rule) {
        option (google.api.http) = {
            put: "/rules"
            body: "rule"
        };
    }
```